### PR TITLE
EPT reader headers/query parameters forwarding for HTTP-based endpoints

### PR DESCRIPTION
**Pull request by connormanning**
Adds `headers` and `query` options to the EPT reader, which will be forwarded to remote endpoints.  The typical use-case here is for authenticated endpoints.

Also adds capability for `filename` values ending with `"ept.json"` to be routed to the EPT reader, which can be more convenient than the alternative `ept://` protocol method.

### CODE REVIEW

File: io/EptReader.cpp [hunk 1]
```diff
@@ -159,6 +161,38 @@ void EptReader::addArgs(ProgramArgs& args)
     args.add("polygon", "Bounding polygon(s) to crop requests",
         m_args->m_polys).setErrorText("Invalid polygon specification. "
             "Must be valid GeoJSON/WKT");
+    args.add("header", "Header fields to forward with HTTP requests",
```
Comment (abellgithub) [inline io/EptReader.cpp]: Ugh to JSON here.  But I know I lose.

**Comment by connormanning** [comment]
The parameters are validated as either `null` or object mappings whose values are all strings [here](https://github.com/PDAL/PDAL/pull/2690/files#diff-6c6aa6fc314618e64694ec420896da89R304), and an error occurs otherwise:

```
pdal info http://na.entwine.io/autzen/ept.json --readers.ept.query="{\"asdf\":42}"
PDAL: readers.ept: Invalid query parameters: expected string->string mapping
```

**Comment by connormanning** [comment]
Or if not an object:

```
pdal info http://na.entwine.io/autzen/ept.json --readers.ept.query=42
PDAL: readers.ept: Invalid query parameters: expected object
```

**Comment by abellgithub** [comment]
Sorry.  I saw that check.  I mean when those query/header parameters are sent downstream.

**Comment by connormanning** [comment]
If the service endpoint errors due to the *contents* of the parameters, for example if a `401` status code were returned, then the error handling is the same as any other endpoint error:
```
PDAL: readers.ept: Could not read from http://my-server.com/whatever/ept.json
```

This is potentially a spot for improvement, but presumably for users opting in to supplying headers/query options for authenticated endpoints, the contents of those options would be the obvious place to look.

**Pull request merged**